src/ion.hpp

```diff
@@ -58,13 +56,13 @@ public :
 
     ion() = default;
 
-    ion(const std::vector<size_type>& idx) :
+    ion(const std::vector<size_type>& idx):
         node_index_{memory::make_const_view(idx)},
         iX_{idx.size(), std::numeric_limits<value_type>::quiet_NaN()},
         eX_{idx.size(), std::numeric_limits<value_type>::quiet_NaN()},
         Xi_{idx.size(), std::numeric_limits<value_type>::quiet_NaN()},
         Xo_{idx.size(), std::numeric_limits<value_type>::quiet_NaN()},
-        valency(0),
+        charge(0),
```

Comment:
It turns out valence is a property of elements rather than ions, and so an ion of a particular element might have more than one possible charge while that element has only the one valence.

So changed to 'charge' just to avoid ambiguity.